File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long reverse_number = 0;
        int MIN = -1*2147483648;
        int MAX = 2147483647;
        while (x != 0) {
            reverse_number *= 10;
            reverse_number += (x % 10);
            if (reverse_number > MAX || reverse_number < MIN) {
                return 0;  // This would cause overflow
            }
            x -= (x % 10);
            x /= 10;
        }
        return reverse_number;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):   
        reversed_x = 0
        count = 0
        negative = False
        if x < 0:
            x = -1 * x
            negative = True
        while x > 0:
            reversed_x = reversed_x * 10 + (x % 10)
            print(reversed_x)
            x /= 10
        
        if reversed_x > 2**31 - 1:
            return 0
        if negative:
            reversed_x *= -1
        return reversed_x
        """
        :type x: int
        :rtype: int
        """
        